src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int RANDOM_LENGHT = 20;
+    private final ColorSupplier randomColor = new ColorSupplier();
+    private final Random random = new Random();
+
+    public int getRandomNumber() {
+        return random.nextInt(RANDOM_LENGHT);
+    }
+
+    public TypeOfFigure getRandomTypeOfFigure() {
+        return TypeOfFigure.values()[random.nextInt(TypeOfFigure.values().length)];
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(Color.WHITE, 10);
+    }
+
+    public Figure getRandomFigure() {
+        int randomHeight = getRandomNumber();
+        int randomSide1 = getRandomNumber();
+        int randomSide2 = getRandomNumber();
+        int randomRadius = getRandomNumber();

```

Comment:
3/4 variables useless in case `CIRCLE`, better to use `getRandomNumber` directly.